gamification-evm-services/src/main/java/io/meeds/evm/gamification/service/EvmContractTransferService.java

```diff
@@ -0,0 +1,169 @@
+package io.meeds.evm.gamification.service;
+
+import io.meeds.evm.gamification.model.EvmTransaction;
+import io.meeds.evm.gamification.model.EvmTrigger;
+import io.meeds.evm.gamification.utils.Utils;
+import io.meeds.gamification.constant.DateFilterType;
+import io.meeds.gamification.constant.EntityStatusType;
+import io.meeds.gamification.model.RuleDTO;
+import io.meeds.gamification.model.filter.RuleFilter;
+import io.meeds.gamification.service.RuleService;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.exoplatform.services.listener.ListenerService;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.math.BigInteger;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static io.meeds.evm.gamification.utils.Utils.EVM_SAVE_ACTION_EVENT;
+
+@Service
+public class EvmContractTransferService {
+
+  private static final Log      LOG = ExoLogger.getLogger(EvmContractTransferService.class);
+
+  @Autowired
+  private EvmTransactionService evmTransactionService;
+
+  @Autowired
+  private ListenerService       listenerService;
+
+  @Autowired
+  private EvmTriggerService     evmTriggerService;
+
+  @Autowired
+  private EvmBlockchainService  evmBlockchainService;
+
+  @Autowired
+  private RuleService           ruleService;
+
+  public void listenEvmContractTransfer(RuleDTO rule) {
```

Comment:
It's not listening but scanning for new transactions on contract since last scan date.

```suggestion
  public void scanForContractTransactions(RuleDTO rule) {
```